src/lib/transaction_snark/transaction_snark.ml

```diff
@@ -5343,6 +5374,51 @@ let%test_module "transaction_snark" =
                 Token_permissions.equal
                   (Not_owned {account_disabled= true})
                   target_account.token_permissions ) ) )
+
+    let%test_unit "disabled accounts cannot send transactions" =
+      Test_util.with_randomness 123456789 (fun () ->
+          Ledger.with_ledger ~depth:ledger_depth ~f:(fun ledger ->
+              let wallets = random_wallets ~n:2 () in
+              let signer = wallets.(0).private_key in
+              let fee_payer_pk = wallets.(0).account.public_key in
+              let source_pk = fee_payer_pk in
+              let receiver_pk = wallets.(1).account.public_key in
+              let fee_token = Token_id.default in
+              let token_id =
+                Quickcheck.random_value Token_id.gen_non_default
+              in
+              let accounts =
```

Comment:
might be a little clearer to name these, like `fee_payer_account0`, so you don't have to pull them out of the array in the asserts below